sdks/python/apache_beam/runners/worker/worker_status.py

```diff
@@ -0,0 +1,148 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Worker status api handler for reporting SDK harness debug info."""
+
+from __future__ import absolute_import
+from __future__ import division
+
+import queue
+import sys
+import threading
+import traceback
+from collections import defaultdict
+
+import grpc
+
+from apache_beam.portability.api import beam_fn_api_pb2
+from apache_beam.portability.api import beam_fn_api_pb2_grpc
+from apache_beam.runners.worker.channel_factory import GRPCChannelFactory
+from apache_beam.runners.worker.worker_id_interceptor import WorkerIdInterceptor
+
+
+def thread_dump():
+  """Get a thread dump for the current SDK worker harness. """
+  # deduplicate threads with same stack trace
+  stack_traces = defaultdict(list)
+  frames = sys._current_frames()  # pylint: disable=protected-access
+
+  for t in threading.enumerate():
+    stack_trace = ''.join(traceback.format_stack(frames[t.ident]))
+    thread_ident_name = (t.ident, t.name)
+    stack_traces[stack_trace].append(thread_ident_name)
+
+  all_traces = ['=' * 10 + 'THREAD DUMP' + '=' * 10]
+  for stack, identity in stack_traces.items():
```

Comment:
Let's add a jira to group threads which have same thread stack for easier analysis and reducing text size.
This can be a starter task for new beam contributors.

Reply:
isn't it already included in this PR?

Reply:
You are right. It's already in this PR. 
We can print names of all the threads along with count so that we don't miss any information.